stan/math/prim/fun/cholesky_factor_free.hpp

```diff
@@ -48,6 +48,22 @@ Eigen::Matrix<value_type_t<T>, Eigen::Dynamic, 1> cholesky_factor_free(
   return x;
 }
 
+/**
+ * Overload of `cholesky_factor_free()` to untransform each matrix
+ * in a standard vector.
+ * @tparam T A standard vector with with a `value_type` which inherits from
+ *  `Eigen::MatrixBase`.
+ * @param x The standard vector to untransform.
+ */
+template <typename T, require_std_vector_t<T>* = nullptr>
+auto cholesky_factor_free(const T& x) {
```

Comment:
Is there a way to do this the same way we do array applies for vectorized functions so there's not so much cut and paste?   I'm thinking the definitions could look as follows.

```
template <typename T, require_std_vector_t<T>* = nullptr>
auto cholesky_factor_free(const T& x) {
  return apply_free([](auto&& xx) {  return cholesky_factor_free(xx); });
}
```

The helper function would then be as follows (it doesn't need to validate the standard vector arg).

```
template <typename T, typename F>
auto apply_free(const T& x, const F& f) {
  std::vector<decltype(f(x[0]))> y(x.size());
  std::transform(x.begin(), x.end(), y.begin(), f);
  return y;
}
```

It'd be even better without having to repeat the requires clause in each function and without the boilerplate lambda abstraction, but I don't see how to get rid of those.

With the generic implementation, we have the type `F` so we can use [`result_of`](https://en.cppreference.com/w/cpp/types/result_of), which would look as follows.

```
std::vector<result_of_t<F, T::value_type>> x_free(x.size());
```

But this needs `F` to be a function type, so it'd only work in the generic version here as far as I can tell.

Reply:
Oh actually I wonder if I can just use the `apply_vector_unary` framework here. Lemme try that

Reply:
Yes! Just updated this, code uses the `apply_vector_unary<>()` framework and is more generic now. We need something similar for the constraints and I can do the same thing there as well

Reply:
Also nice!   The constraining transforms are trickier because of the side effects.  Not to mention a different PR.